app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,215 @@
+    border: `1px solid ${theme.palette.divider}`,
+
+    borderRadius: theme.shape.borderRadius,
+    "& > div": {
+      padding: theme.spacing(3),
+      borderRadius: theme.shape.borderRadius,
+    },
+  },
+}));
+
+const Preview = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const { sourceId, mappingId, attributeId } = useParams<{
+    sourceId?: string;
+    mappingId?: string;
+    attributeId?: string;
+  }>();
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    {
+      id: mappingId ?? "",
+    },
+    {
+      skip: !Boolean(mappingId),
+    }
+  );
+  const [fhirInstance, setFhirInstance] = useState<
+    typeof patientJson | undefined
+  >(undefined);
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+
+  const FhirIconCell = () => (
+    <IconButton size="small" className={classes.icon}>
+      <Icon
+        icon={IconNames.FLAME}
+        iconSize={15}
+        className={classes.iconFlame}
+      />
+    </IconButton>
+  );
+
+  const handleFhirIconClick = (index: number) => {
+    const primarykey = mapping?.primary_key_table;
+    if (primarykey) {
+      const indexPrimaryKey = exploredData.fields.indexOf(primarykey);
+      console.log(exploredData.rows[index]?.[indexPrimaryKey]);
+      setFhirInstance(patientJson);
+    }
+  };
+
+  return (
+    <Container className={classes.container} maxWidth="xl">
+      <Link
+        className={classes.link}
+        variant="h5"
+        color="textSecondary"
+        to={
+          attributeId
+            ? `/sources/${sourceId}/mappings/${mappingId}/attributes/${attributeId}`
+            : `/sources/${sourceId}/mappings/${mappingId}`
+        }
+        gutterBottom
+      >
+        <ArrowBack />
+        {t("back")}
+      </Link>
+      <TableContainer className={classes.table}>
+        <Table size="small">
+          <TableHead>
+            <TableRow className={classes.cellsTitle}>
+              <TableCell className={classes.cells}></TableCell>
+              {exploredData.fields.map((field) => (
+                <TableCell className={classes.cells} key={uuid()}>
+                  {field}
+                </TableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {exploredData.rows.map((columnData, index) => (
+              <TableRow hover key={uuid()} className={clsx(classes.rowBorder)}>
+                <TableCell
+                  className={clsx(classes.fhirIconCell, classes.cellsTitle)}
+                  onClick={() => handleFhirIconClick(index)}
+                >
+                  <FhirIconCell />
+                </TableCell>
+                {columnData.map((an, i) => (
+                  <TableCell className={classes.cells} key={i}>
+                    {an}
+                  </TableCell>
+                ))}
+              </TableRow>
+            ))}
+          </TableBody>
+          <TableFooter></TableFooter>
```

Comment:
Is there supposed to be something inside this `TableFooter` component ?

Reply:
maybe for the pagination ? we could add a comment

Reply:
It was pagination but I deleted it ... I forgot the TableFooter component